src/components/TodoList.tsx

```diff
@@ -0,0 +1,34 @@
+import { useContext } from 'react';
+import { TodoContext } from './TodoContext';
+import { TodoInfo } from './TodoInfo';
+import { Query } from '../types/Query';
+
+type Props = {
+  query: Query;
+};
+
+export const TodoList: React.FC<Props> = ({ query }) => {
+  const { todos } = useContext(TodoContext);
+  let todosToRender = [...todos];
+
+  switch (query) {
+    case Query.All:
+      todosToRender = [...todos];
+
+      break;
+    case Query.Active:
+      todosToRender = [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
      todosToRender = todos.filter(todo => !todo.completed);
```
you don't need to make a copy here, 'cause the filter method doesn't mutate the array